Database structure allowing different databases to be used

takase1121 opened the pull request:
Converted `HarunaStore.js` into a common interface. Users might be able to just install a module to use alternative databases like SQLite, MySQL, Mongo, Postgres, etc.
 This may allow them to integrate them within their application easier (use the same database as their app uses.)
Not only that, by using a non-level backed store, Haruna may be able to run in clusters / allowing other people to access the DB

takase1121 committed:
> Changed database structure
- added base file for a custom db implementation
- adapted original Haruna Store for the new structure
- added config.js* to gitignore

Deivu commented:
Probably have a guideline on how to extend the store I would say.
Examples as well and this would be good

takase1121 commented:
busy rn soon

takase1121 committed:
More info
- added implementation document
- added license on HarunaStore.js for easier copying

takase1121 committed:
fix typo

takase1121 commented:
This PR will be closed and a new one will be opened later. There is a lot more to do.
